### PR TITLE
fix(kit): `Tree` fix control chevron rotation programmatically

### LINKED ISSUE #6834: 🐞 - cannot control chevron rotation programatically in treeview component
### Playground Link

https://taiga-ui.dev/components/tree

### Description

When using programmatic control to define which nodes are expanded, the chevron does not update.

In this video from the taiga website, notice that the chevron does not rotate when the node is expanded / collapsed programatically, but does rotate when toggled by clicking

Peek 2024-02-20 04-59

### Angular version

16

### Taiga UI version

3.66.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [ ] Windows
- [X] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by 0x083**
Closes #6834

### Description
I just copied the change detection logic from the TreeItem component and at least it works.

### CODE REVIEW

File: projects/kit/components/tree/components/tree-item-content/tree-item-content.component.ts [hunk 1]
```diff
@@ -18,14 +20,29 @@ import {TUI_TREE_CONTROLLER} from '../../misc/tree.tokens';
...
+    protected readonly expanded$ = this.change$.pipe(
+        startWith(null),
+        map(() => this.isExpanded),
+    );
+
+    public ngDoCheck(): void {
+        this.checkChanges();
+    }
+
+    public checkChanges(): void {
```
Comment (waterplea) [inline projects/kit/components/tree/components/tree-item-content/tree-item-content.component.ts:42]: This method is unnecessary

File: projects/kit/components/tree/components/tree-item-content/tree-item-content.component.ts [hunk 2]
```diff
@@ -18,14 +20,29 @@ import {TUI_TREE_CONTROLLER} from '../../misc/tree.tokens';
...
     );
 
+    private readonly change$ = new Subject<void>();
+
     protected readonly icons = inject(TUI_COMMON_ICONS);
     protected readonly context = inject<TuiTreeItemContext>(POLYMORPHEUS_CONTEXT);
 
+    protected readonly expanded$ = this.change$.pipe(
+        startWith(null),
+        map(() => this.isExpanded),
```
Comment (waterplea) [inline projects/kit/components/tree/components/tree-item-content/tree-item-content.component.ts:35]: Please add distinctUntilChanges

**Comment by 0x083** [comment]
Necessary changes have been made

**Review by splincode — APPROVED** [review]

**Review by waterplea — APPROVED** [review]
Thank you for this PR. If you want it in the current version, we also need the same changes in v3.x branch

**Comment by 0x083** [comment]
> Thank you for this PR. If you want it in the current version, we also need the same changes in v3.x branch

Can I just edit the PR and change the target branch from master to v3.x?

**Comment by waterplea** [comment]
> > Thank you for this PR. If you want it in the current version, we also need the same changes in v3.x branch
> 
> Can I just edit the PR and change the target branch from master to v3.x?

As you wish. We'll cherry pick it to main.

**Pull request merged**